controllers/envoy/envoyconfig_controller.go

```diff
@@ -60,34 +58,21 @@ func (r *EnvoyConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 		return ctrl.Result{}, err
 	}
 
-	// Add finalizer for this CR
-	if !contains(ec.GetFinalizers(), envoyv1alpha1.EnvoyConfigFinalizer) {
-		r.Log.Info("Adding Finalizer for the EnvoyConfig")
-		if err := r.addFinalizer(ctx, ec); err != nil {
-			r.Log.Error(err, "Failed adding finalizer for envoyconfig")
-			return ctrl.Result{}, err
-		}
+	// Set defaults.
+	// TODO: remove this when wwe migrate to CRD v1 api as we will
+	// be able to set defauls directly in the CRD definition.
+	if err := r.reconcileSpecDefaults(ctx, ec); err != nil {
+		return ctrl.Result{}, err
 	}
 
-	// Check if the EnvoyConfig instance is marked to be deleted, which is
-	// indicated by the deletion timestamp being set.
-	if ec.GetDeletionTimestamp() != nil {
-		if contains(ec.GetFinalizers(), envoyv1alpha1.EnvoyConfigFinalizer) {
-			r.finalizeEnvoyConfig(ec.Spec.NodeID)
-			r.Log.V(1).Info("Successfully cleared ads server cache")
-			// Remove memcachedFinalizer. Once all finalizers have been
-			// removed, the object will be deleted.
-			controllerutil.RemoveFinalizer(ec, envoyv1alpha1.EnvoyConfigFinalizer)
-			err := r.Client.Update(ctx, ec)
-			if err != nil {
-				return ctrl.Result{}, err
-			}
-		}
-		return ctrl.Result{}, nil
+	// This must be done before any other revision related code due to the addition of support for
+	// envoy API v3 which comes with the addition of a new label to identify revisions belonging
+	// to each api version. The first time this version of the controller runs, it will set the
+	// required labels. This will also help with future labels that get added.
+	if err := r.reconcileRevisionLabels(ctx, ec); err != nil {
+		return ctrl.Result{}, err
 	}
```

Comment:
The EnvoyConfig controller relies on labels to get the EnvoyConfigRevision resources relevant to it. As new labels have been added, this needs to be reconciled so when upgrading from an older marin3r version, the labels get added to all EnvoyConfigRevision objects.